### ISSUE TITLE
Heatmap using categorical colormaps with GR backend

### ISSUE THREAD

**Issue by diegozea** [JuliaPlots/Plots.jl#1441]
Hi! I've found that the GR backend doesn't use a categorical colormap for categorical values:

![imagen](https://user-images.githubusercontent.com/2822757/37534053-f1b2343a-2943-11e8-89ee-16ca008c7508.png)

The code of the example is the same that the one that tbreloff commented on 26 Apr 2016 in this issue https://github.com/JuliaPlots/Plots.jl/issues/204 :
```julia
using Plots
gr()
n, m = 15, 4
xs = [string("x",i) for i in 1:n]
ys = [string("y",i) for i in 1:m]
z = rand(["A","B","C"], n, m)
heatmap(xs, ys, z, yflip=true, c=[:red, :blue, :green], l=(1,:black),
grid=false, axiscolor=nothing, border=nothing, size=(600,300))
```

**Comment by mkborregaard** [comment]
This is true, categorical colormaps aren't yet supported. https://github.com/JuliaPlots/PlotUtils.jl/issues/3 . Would be great, though, but it would require developing it.
